doc/users/whats_new.rst

```diff
@@ -19,6 +19,15 @@ New in matplotlib 2.0
 
    matplotlib 2.0 supports Python 2.7, and 3.4+
 
+The behavior of AxesStack re-add key changes
```

Comment:
I'm not sure `AxesStack` is what should be documented here; it's an internal (if not private) implementation detail and what the user really sees is that `Figure.add_axes` (and `plt.axes`?) no longer has this behaviour. "What's new" should be about user-visible changes. This specific information could probably go in `api/api_changes`.